react-front-end/tsrc/modules/ViewerModule.ts

```diff
@@ -248,3 +248,190 @@ export const buildLightboxNavigationHandler = (
+ * message and right is 'AttachmentAndViewerDefinition'. However, the final result is an Either where left is a string
+ * which accumulates all the error messages and right is an array of 'AttachmentAndViewerDefinition'.
+ *
+ * @param attachments Attachments provided to find their viewer configurations.
+ * @param uuid UUID of the Item which the attachments belong to.
+ * @param version Version of the Item which the attachments belong to.
+ * @param getViewerDetails Function called to retrieve viewer detail for each attachment.
+ */
+export const buildAttachmentsAndViewerDefinitions = async (
+  attachments: OEQ.Search.Attachment[],
+  uuid: string,
+  version: number,
+  getViewerDetails: (
+    mimeType: string
+  ) => Promise<OEQ.MimeType.MimeTypeViewerDetail>
+): Promise<E.Either<string[], AttachmentAndViewerDefinition[]>> => {
+  const either: E.Either<string, AttachmentAndViewerDefinition>[] =
+    await Promise.all(
+      attachments.map(async (originalAttachment) => {
+        const { brokenAttachment } = originalAttachment;
+        // Broken attachments don't have viewer configuration so we just return the attachment.
+        if (brokenAttachment) {
+          return E.right({ attachment: originalAttachment });
+        }
+
+        const attachment = updateAttachmentForCustomInfo(originalAttachment);
+        const { mimeType } = attachment;
+
+        return await pipe(
+          mimeType,
+          O.fromNullable,
+          O.foldW(
+            // If MIME type is undefined, we just need a TaskEither which returns undefined as ViewerId, otherwise we
+            // call the provided function to retrieve ViewerId.
+            () =>
+              TE.right<string, OEQ.MimeType.ViewerId | undefined>(undefined),
+            (m) =>
+              pipe(
+                TE.tryCatch(() => getViewerDetails(m), String),
+                TE.map(
+                  (resp: OEQ.MimeType.MimeTypeViewerDetail) => resp?.viewerId
+                )
+              )
+          ),
+          // Then we map ViewerId to AttachmentAndViewerDefinition.
+          TE.map((viewerId: OEQ.MimeType.ViewerId | undefined) => ({
+            attachment,
+            viewerDefinition: getViewerDefinitionForAttachment(
+              uuid,
+              version,
+              attachment,
+              viewerId
+            ),
+          }))
+        )();
+      })
+    );
+
+  const lefts = A.lefts(either);
+  const rights = A.rights(either);
+  return A.isEmpty(lefts) ? E.right(rights) : E.left(lefts);
+};
+
+/**
+ * Build Lightbox entries based on a list of attachments and viewer definitions
+ *
+ * @param attachmentsAndViewerDefinitions A list of attachments and their viewer definitions.
+ */
+export const buildLightboxEntries = (
+  attachmentsAndViewerDefinitions: AttachmentAndViewerDefinition[]
+): LightboxEntry[] =>
+  pipe(
+    attachmentsAndViewerDefinitions,
+    A.map(({ attachment: { id, description, mimeType }, viewerDefinition }) =>
```

Comment:
Might be possible here to utilise `flow`:

```suggestion
export const buildLightboxEntries = (
  attachmentsAndViewerDefinitions: AttachmentAndViewerDefinition[]
): LightboxEntry[] =
  flow(
    A.map(({ attachment: { id, description, mimeType }, viewerDefinition }) =>
```